src/main/java/de/uniba/dsg/serverless/benchmark/BenchmarkExecutor.java

```diff
@@ -0,0 +1,119 @@
+package de.uniba.dsg.serverless.benchmark;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.URL;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import com.google.common.io.CharStreams;
+
+import de.uniba.dsg.serverless.model.SeMoDeException;
+
+public class BenchmarkExecutor {
+
+	private static URL url;
+	private int numberOfRequests;
+
+	public BenchmarkExecutor(URL url, int numberOfRequests) {
+		BenchmarkExecutor.url = url;
+		this.numberOfRequests = numberOfRequests;
+	}
+
+	/**
+	 * Executes a benchmark in one of three modes:<br>
+	 * {@link BenchmarkMode#CONCURRENT}<br>
+	 * Executes the function numberOfRequests times without delay.
+	 * <p>
+	 * {@link BenchmarkMode#SEQUENTIAL_INTERVAL}<br>
+	 * Executes the function at the given fixed rate. The first execution will
+	 * commence immediately. Further ones will be executed at <math>(T +
+	 * delay)</math>, <math>(T + 2 * delay)</math> and so on. If a function
+	 * execution exceeds the delay, further executions are delayed and not executed
+	 * concurrently.
+	 * <p>
+	 * {@link BenchmarkMode#SEQUENTIAL_INTERVAL}<br>
+	 * Executes the function at the given delay after the function execution time.
+	 * The first execution will commence immediately. Furhter ones will be executed
+	 * at <math>(T + executionTime1 + delay)</math> and so on.
+	 * 
+	 * @param delay
+	 *            delay between function executions (0 for concurrent)
+	 * @param mode
+	 *            Mode of the Benchmark
+	 * @throws SeMoDeException
+	 */
+	public void executeBenchmark(int delay, BenchmarkMode mode) throws SeMoDeException {
+		ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+		switch (mode) {
+		case CONCURRENT:
+			for (int i = 0; i < numberOfRequests; i++) {
+				triggerFunction();
+			}
+			return;
+		case SEQUENTIAL_INTERVAL:
+			System.out.println("start");
+			ScheduledFuture<?> handle = executorService.scheduleAtFixedRate(BenchmarkExecutor::triggerFunction, 0,
+					delay, TimeUnit.SECONDS);
```

Comment:
Exceptions cannot be detected at this point. However, they are important since the function execution must be guaranteed.